saltgui/static/scripts/routes/keys.js

```diff
@@ -6,23 +6,57 @@ class KeysRoute extends PageRoute {
     this.jobsLoaded = false;
 
     this._updateKeys = this._updateKeys.bind(this);
+    this._updateKeysFingerprint = this._updateKeysFingerprint.bind(this);
   }
 
   onShow() {
     const keys = this;
+    const p1 = keys.router.api.getKeys();
+    const p2 = keys.router.api.getKeysFingerprint();
+
+    Promise.all([p1, p2])
+      .then(function(data){
+        // process result of 1st promise
+        keys._updateKeys(data[0]);
+        // process result of 2nd promise
+        keys._updateKeysFingerprint(data[1]);
+      });
     return new Promise(function(resolve, reject) {
       keys.resolvePromise = resolve;
       if(keys.keysLoaded && keys.jobsLoaded) resolve();
-      keys.router.api.getMinions().then(keys._updateMinions);
-      keys.router.api.getKeys().then(keys._updateKeys);
       keys.router.api.getJobs().then(keys._updateJobs);
       keys.router.api.getJobsActive().then(keys._runningJobs);
     });
   }
 
+  _updateKeysFingerprint(data) {
+    const keys = data.return[0].data.return;
+
+    for(const property of Object.keys(keys)) {
+      if(property === "local") continue;
```

Comment:
Just noticed... The answer has exactly 2 keys: `local` and `minions`. Just pick the `minions` part instead of looping through all available keys.

Reply:
it has more to be strict:
- local - master
- minion - accepted minions keys
- minions_rejected - rejected minions keys
- minions_pre - not accepted minions keys